docs/sources/shared/wal-data-retention.md

```diff
@@ -0,0 +1,79 @@
+---
+aliases:
+- /docs/agent/shared/wal-data-retention/
+- /docs/grafana-cloud/agent/shared/wal-data-retention/
+- /docs/grafana-cloud/monitor-infrastructure/agent/shared/wal-data-retention/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/shared/wal-data-retention/
+canonical: https://grafana.com/docs/agent/latest/shared/wal-data-retention/
+description: Shared content, information about data retention in the WAL
+headless: true
+---
+
+The `prometheus.remote_write` component reuses the Prometheus Write-Ahead Log
+(WAL) implementation for resiliency against network outages and to avoid the
+prohibitively expensive in-memory buffering of all metric samples. The
+component buffers the received metrics on-disk, in a WAL per configured
+endpoint. The queue shards can use the WAL after the network outage is resolved
+and flush the buffered metrics to the endpoints.
+
+The WAL records metrics in 128MB-files called segments. To avoid having a WAL
+that grows on-disk indefinitely, the component _truncates_ its segments on a
+set interval.
+
+On each truncation, the WAL deletes references to series that are no longer
+present and also _checkpoints_ roughly the oldest two thirds of the segments
+written to it. A checkpoint means that the WAL only keeps track of the unique
```

Comment:
What does "roughly" mean in this context? It's exactly 2/3rds of segment files, rounded down to the nearest integer :)

Reply:
Same point for line 66.... there we have it saying "approximately 2/3rds. Do we need to be more explicit and state 2/3rds rounded down?

Reply:
A weasel word to escape describing the calculation 😅  will rephrase this.